logging/google/cloud/logging/_gax.py

```diff
@@ -444,13 +453,49 @@ def metric_delete(self, project, metric_name):
             raise
 
 
+def _parse_log_entry(entry_pb):
+    """Special helper to parse ``LogEntry`` protobuf into a dictionary.
+
+    The ``proto_payload`` field in ``LogEntry`` is of type ``Any``. This
+    can be problematic if the type URL in the payload isn't in the
+    ``google.protobuf`` registry. To help with parsing unregistered types,
+    this function will remove ``proto_payload`` before parsing.
+
+    :type entry_pb: :class:`.log_entry_pb2.LogEntry`
+    :param entry_pb: Log entry protobuf.
+
+    :rtype: dict
+    :returns: The parsed log entry. The ``protoPayload`` key may contain
+              the raw ``Any`` protobuf from ``entry_pb.proto_payload`` if
+              it could not be parsed.
+    """
+    try:
+        return MessageToDict(entry_pb)
+    except TypeError:
```

Comment:
@lukesneeringer @tseaver I had mentioned that I wanted to avoid flow control with exceptions here, but there is no real path offered by `google.protobuf`.

I started digging in with the debugger at:

```python
>>> from google.protobuf import descriptor_pool
>>> pool = descriptor_pool.Default()
>>> type_name = 'google.bigtable.admin.v2.UpdateClusterMetadata'
>>> pool.FindMessageTypeByName(type_name)
```

but it turns out `google.protobuf` also does flow control with exceptions (or it dives into C extensions)